Google analytics

casuwu opened the pull request.

omnipounder committed:
Adding feedback button on side nav / issue #401

omnipounder committed:
Adding Canny logo to align with #401 ask

omnipounder committed:
editing icon

omnipounder committed:
push to ga user stake/unstake event init

omnipounder committed:
clean

omnipounder committed:
capture user bond event init

omnipounder committed:
fix poiner

omnipounder committed:
capture user redeem bond event init

casuwu committed:
removing analytic push to google

discussing service to use today sep20

casuwu committed:
adding approved bool and txHash to redeem,bond,stake,unstake event capture

casuwu committed:
init segment analytics, push event redeem/bond/unstake/stake

casuwu commented:
User Analytics Project

1. Integrate segment analytics
2.  Capture redeem, bond, stake, unstake events, and push to segment (based on requirements detailed here: https://docs.google.com/document/d/1c48Jn8WP9nY5JahMtYBd0NojL05GdUXkUa8DMfSSO1s/edit)
3. Add feedback button ( currently hosted on staging issue #401)

Open to feedback/suggestions

Zayen-X commented on src/hooks/useSegmentAnalytics.js:
```diff
@@ -0,0 +1,57 @@
+export default function useSegmentAnalytics() {
+  !(function () {
```
why do you need a self  invoking anonymous function? i never liked the syntax of this shit.

casuwu replied:
part of the docs for segment, just straight copy pasted. Not sure how else we could go about it

Zayen-X commented on src/hooks/useSegmentAnalytics.js:
```diff
@@ -0,0 +1,57 @@
+        };
+        analytics._writeKey = "nhYamX4lIWdDsEvqHRonx1bbwFEcA2TD";
+        analytics.SNIPPET_VERSION = "4.15.3";
+        analytics.load("nhYamX4lIWdDsEvqHRonx1bbwFEcA2TD");
```
I think we  should move these into .env files and have different keys for staging + prod. Lets  get @appleseed-iii (#415) in therrr.

casuwu replied:
absolutely a good idea!

Zayen-X commented on src/hooks/userAnalyticHelpers.js:
```diff
@@ -0,0 +1,7 @@
+// Pushing data to segment analytics
```
technically this isn't a hook. heh. 
You could refactor it, and have the segment analytics initialize through through a `useEffect(()=>{},[])` and maintain the state for the `analytics` object. probably a better pattern imo.

casuwu replied:
Okay! we can sync to hash out precisely how to get it done, happy to iron it out. ty for the review!

casuwu committed:
API Key pattern edit

Commit:
linter err

appleseed-iii commented on src/helpers/Environment.ts:
```diff
@@ -0,0 +1,117 @@
+        ) {
+            return EnvHelper.env.REACT_APP_PROD_SEGMENT_KEY
+        } else if (EnvHelper.env.REACT_APP_TESTNET_SEGMENT_KEY){
+            return EnvHelper.env.REACT_APP_TESTNET_SEGMENT_KEY
```
In `.env` I would make the segment key the exact same VAR name. So `REACT_APP_SEGMENT_KEY="whatever"`, rather than separating var names by PROD vs TESTNET. The ENV already does that inherently.

appleseed-iii replied:
also, rather than manually copying this file from develop into your branch you should probably just merge develop into your branch. By copying it you ended up with a lot of `prettier` errors: 
![image](https://user-images.githubusercontent.com/80423742/134545321-a8e7b486-2495-44d0-8e19-6b919bc28f13.png)

casuwu replied:
yeah, just merged in develop, linter errors should be out

appleseed-iii commented on .sample.env:
```diff
@@ -1 +1,31 @@
+REACT_APP_SELF_HOSTED_WEBSOCKETS="ws://ipAddress:port"
+
+#
+REACT_APP_TESTNET_SEGMENT_KEY=""
```
If the dev leaves this key blank in `development` environment we won't have a crash / error, right? Since we don't need to track dev usage via segment they **should** be able to leave it blank.

appleseed-iii replied:
this should be `REACT_APP_SEGMENT_KEY`.
TESTNET vs PROD doesn't need to be carried separately since that's handled by the ENV inherently.

casuwu replied:
Gotcha. Env should be fine so long as there is an empty string passed at least. Segment won't initialize but thats alright, as you say dev tracking isn't necessary apart for testing purposes. 

Pushing a change to alter REACT_APP_TESTNET to REACT_APP_SEGMENT_KEY

casuwu committed:
cont

casuwu committed:
invoking env helper func to retrieve api key

casuwu committed:
Merge branch 'develop' into googleAnalytics

appleseed-iii commented on src/slices/BondSlice.ts:
```diff
@@ -190,22 +190,33 @@ export const bondAsset = createAsyncThunk(
+      address: address,
+      value: value,
+      type: "Bond",
+      bondName: bond,
```
this should probably be `bond.displayName`, right?

casuwu replied:
correct! pushing a change.

appleseed-iii commented on src/slices/BondSlice.ts:
```diff
@@ -232,20 +243,32 @@ export const redeemBond = createAsyncThunk(
+    let uaData = {
+      address: address,
+      type: "Redeem",
+      bondName: bond,
```
`bond.displayName`?

casuwu replied:
pushed a change

appleseed-iii commented on src/helpers/Environment.ts:
```diff
@@ -10,9 +10,19 @@ export class EnvHelper {
+    if (EnvHelper.env.NODE_ENV !== "development" && EnvHelper.env.REACT_APP_PROD_SEGMENT_KEY) {
+      return EnvHelper.env.REACT_APP_PROD_SEGMENT_KEY;
+    } else if (EnvHelper.env.REACT_APP_TESTNET_SEGMENT_KEY) {
+      return EnvHelper.env.REACT_APP_TESTNET_SEGMENT_KEY;
```
elsewhere in my review, too, you can just use one `REACT_APP_SEGMENT_KEY` & then you don't need all this IF statement logic.

casuwu committed:
Cleaning code to address comments

appleseed-iii committed:
update segment analytics

1. remove extra .sample.env (replaced by .env.example)
2. react-ify useSegmentAnalytics / make it a hook

Potential future work:
- see notes in `useSegmentAnaltyics.js`: `location.pathname` does not change with IPFS sites
  so we may need to manually pass in `location.hash`

appleseed-iii committed:
cleanup: don't load segment unless it's present in env, move userAnalyticHelpers into helpers

appleseed-iii reviewed (approved):
LGTM. Before merging can you test the bond, stake & page loading actions with your data folks working on this project w you: hippo, edgecaser, etc. ?

Just to verify that the data is formatted the way they want it. My main concern is in the `location.pathname` (see the notes in the useEffect of useSegmentAnaltyics). I couldn't tell from Segment's docs how the default `analytics.page()` call works. If it only sends `location.pathname` then our current implementation will not work & we will need to add `location.hash`.

casuwu committed:
adding check for env var

casuwu committed:
fixing env format

Zayen-X commented on src/hooks/useSegmentAnalytics.js:
```diff
@@ -0,0 +1,90 @@
+
+  React.useEffect(() => {
+    if (SEGMENT_API_KEY && SEGMENT_API_KEY.length > 1) {
+      initSegmentAnalytics();
```
This is beautiful.

Zayen-X commented on src/hooks/useSegmentAnalytics.js:
```diff
@@ -0,0 +1,90 @@
+  const [loadedSegment, setLoadedSegment] = useState(false);
+  const location = useLocation();
+
+  React.useEffect(() => {
```
nitpick: could just import `useEffect`.

Zayen-X reviewed (approved):
LGTM going to merge this into develop.
